app/services/record_declarations/base.rb

```diff
@@ -119,6 +120,11 @@ def validate_milestone!
       end
     end
 
+    def validate_participant_state!
+      last_state = participant_profile_states.where("created_at < ?", declaration_date).order(:created_at).last
+      raise ActionController::ParameterMissing, I18n.t(:declaration_on_incorrect_state) unless last_state&.state.nil? || last_state&.state == "active"
```

Comment:
```suggestion
      raise ActionController::ParameterMissing, I18n.t(:declaration_on_incorrect_state) unless last_state&.state.nil? || last_state.active?
```

We can use enum helpers here, and the first condition allow us to be unsafe in the latter.

Reply:
yeah, I also don't think we can have nil state here

Reply:
Shouldn't the state checking be a method on the profile as opposed to a reach through?